partner_delivery_link/models/res_partner.py

```diff
@@ -0,0 +1,36 @@
+from odoo import fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    delivery_count = fields.Integer(compute="_compute_delivery_count")
+
+    def _compute_delivery_count(self):
+        delivery_obj = self.env["stock.picking"]
+        for partner in self:
+            partner.delivery_count = delivery_obj.search_count(
+                [
+                    ("partner_id", "=", partner.id),
+                    ("picking_type_id.code", "=", "outgoing"),
+                ]
+            )
+
+    def action_view_partner_delivery(self):
+        delivery_orders = self.env["stock.picking"].search(
+            [
+                ("partner_id", "=", self.id),
+                ("picking_type_id.code", "=", "outgoing"),
+            ]
+        )
+        delivery_tree_view = self.env.ref("stock.vpicktree")
+        delivery_form_view = self.env.ref("stock.view_picking_form")
```

Comment:
for this, domain in both methods is same, so maybe it is worth to consider to have fields like:
M2X delivery_order_ids and integer delivery_count, use one computed method (_compute_delivery_count) for both fields and then:
- we search by this domain inside compute method and assign records  to delivery_order_ids
- delivery_count would be just len(self.delivery_count_ids). 

With this approach we are sure that both fields which rely on same recordset are computed in one method.
Then in domain returned by action we refer to delivery_order_ids.ids

Reply:
Thank you for reviewing the code and I will modify the code as per your suggesion